aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -81,12 +81,25 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ empty ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Tensor empty(IntList size, const TensorOptions& options) {
+#ifndef USE_TH_SIZE_ZERO_DIM
   return options.type().tensor(size);
+#else
+  if (options.type().is_sparse() && size.size() == 0) {
```

Comment:
I don't get why you need to special case this -- shouldn't options.type().tensor(size) just do the right thing?  This looks like you are still expecting people to pass in size.size() == 0 when they mean an empty tensors, but why?

Reply:
This code path is called by `x = torch.sparse.DoubleTensor()` which passes an empty size to this function. Since we don't differentiate how the empty constructor is dispatched in higher level, we probably need special handling for sparse tensor here.

Reply:
but it's also called outside of the code path for `torch.sparse.DoubleTensor()` -- can we change `torch.sparse.DoubleTensor()` to pass in the correct shape?